### DISCUSSION TITLE
NestedFormProvider

### DISCUSSION THREAD

**Discussion by bglgwyng** [react-hook-form/react-hook-form#8533] [Ideas]
Consider the following example

```tsx
const App = () => {
  const methods = useForm(/* ... */);

  return (
    <FormProvider {...methods}>
      <NestedFormProvider path="x.y">
        <A />
      </NestedFormProvider>
    </FormProvider>);
}
```

```tsx
const A = () => {
  const { register } = useFormContext()
  return <input {...register("name")} />
}
```

`NestedFormProvider` makes the children inside nested to `x.y`.
Then the `input` in `A` is registered to `x.y.name` in the form in `App`.
Other methods like `errors` and `setValue` also get nested to `x.y`.
component `A` becomes more reusable. It can now be embedded into any other form that contains the same structure of itself.

I think that there have been similar proposals, but I can't find one on this discussion page.
The most relevant one I saw is https://github.com/react-hook-form/react-hook-form/discussions/7354.
It seems that others have faced the same kind of problems.
However, the approach in the discussion above requires programmers to write path joining code every time.
If react-hook-form provides built-in support for this problem, it would be handier.

Do we have any?

**Comment by bluebill1049** [comment]
No nested support at the moment.

**Reply by bglgwyng** [reply]
Could you tell me more details?
I wonder if this kind of usage is common among react-hook-form users.
If then, what prevents you from introducing built-in support?

Do the solutions that have been suggested, including this one, not look good enough for maintainers?
Or do maintainers consider this problem as not an issue worth to be resolved?
